garak/payloads.py

```diff
@@ -172,49 +169,63 @@ def _scan_payload_dir(self, dir) -> dict:
     def _refresh_payloads(self) -> None:
         """Scan resources/payloads and the XDG_DATA_DIR/payloads for
         payload objects, and refresh self.payload_list"""
-        self.payload_list = self._scan_payload_dir(PAYLOAD_DIR)
+        self.__class__.payload_list = self._scan_payload_dir(PAYLOAD_DIR)
 
     def search(
         self, types: Union[List[str], None] = None, include_children=True
     ) -> Generator[str, None, None]:
         """Return list of payload names, optionally filtered by types"""
-        for payload in self.payload_list:
+        for payload in self.__class__.payload_list:
             if types is None:
                 yield payload
             else:
                 if include_children is False:
                     matches = [
                         payload_type == type_prefix
-                        for payload_type in self.payload_list[payload]["types"]
+                        for payload_type in self.__class__.payload_list[payload][
+                            "types"
+                        ]
                         for type_prefix in types
                     ]
                 else:
                     matches = [
                         payload_type.startswith(type_prefix)
-                        for payload_type in self.payload_list[payload]["types"]
+                        for payload_type in self.__class__.payload_list[payload][
+                            "types"
+                        ]
                         for type_prefix in types
                     ]
                 if any(matches):
                     yield payload
 
+    @staticmethod
+    def _load_payload(
+        name: str, path: Union[str, pathlib.Path, None] = None
+    ) -> PayloadGroup:
+        if path is None:
+            path = PAYLOAD_DIR / f"{name}.json"
+        return PayloadGroup(name, path)
+
```

Comment:
I think I was unclear with this statement:

> Lets consider making the exposed entry point for load_payload be the interface to the Director.

I was thinking that `garak.payloads` module could expose the `Director().search()` capability as a module level method:
``` Python
import garak.payloads

probe_payloads = payloads.search("slur_terms_en").payloads

# or

probe_payloads = payloads.load("slur_terms_en").payloads
```

The `Director` implementation inside `garak.payloads` could then mimic the `PluginCache` class:
``` Python
@staticmethod
def search(types: Union[List[str], None] = None, include_children=True) -> PayloadGroup:
    return Director.getInstance().search(types, include_children)
```

Not a blocker just a direction I think would be easier to consume.

Reply:
Oh, that's pretty neat. Yeah, I like it. Thanks for clarifying.

Reply:
in https://github.com/leondz/garak/pull/933/commits/f8d4d89366fa048ccb76c4cf9230ec13027c519c